omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/support/openmrs2_0/NameSupport2_0.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    public static NameSupport2_0 getInstance() {
+        synchronized (NameSupport2_0.class) {
+            if (singleton == null)
+                singleton = new NameSupport2_0();
+        }
+
+        singleton.init();
+        return singleton;
+    }
+
+    private void init() {
+        if (!this.initialized) {
+            Context.getAdministrationService().addGlobalPropertyListener(singleton);
+            String layoutTemplateXml = getNameFormat();
+            this.setNameTemplate(layoutTemplateXml);
+            List<String> specialTokens = new ArrayList<String>();
+            specialTokens.add("familyName");
+            specialTokens.add("middleName");
+            specialTokens.add("givenName");
+            this.setSpecialTokens(specialTokens);
+            this.initialized = true;
+        }
+    }
+
+    @Override
+    public String getDefaultLayoutFormat() {
+        return this.defaultLayoutFormat;
+    }
+
+    public void setNameTemplate(List<NameTemplate2_0> nameTemplates) {
+        this.layoutTemplates = nameTemplates;
+        this.setDefaultLayoutFormat(this.layoutTemplates.get(0).getCodeName());
+    }
+
+    public List<NameTemplate2_0> getNameTemplate() {
+        if (this.layoutTemplates == null)
+            try {
+                String xml = getNameFormat();
+                this.setNameTemplate(xml);
+            } catch (Exception ignored) {
+            }
+
+        return this.layoutTemplates;
+    }
+
+    @Override
+    public boolean supportsPropertyName(String s) {
+        return getNameFormat().equals(s);
+    }
+
+    @Override
+    public void globalPropertyChanged(GlobalProperty globalProperty) {
+        if (getNameFormat().equals(globalProperty.getProperty())) {
+            try {
+                this.setNameTemplate(globalProperty.getPropertyValue());
+            } catch (Exception var3) {
+                log.error("Error in new xml global property value", var3);
+                this.setNameTemplate((List)(new Vector()));
+            }
+        }
+    }
+
+    private void setNameTemplate(String xml) {
+        NameTemplate2_0 nameTemplate;
+        try {
+            nameTemplate = Context.getSerializationService().getDefaultSerializer().deserialize(xml, NameTemplate2_0.class);
+        } catch (SerializationException var4) {
+            log.error("Error in deserializing name template", var4);
+            nameTemplate = new NameTemplate2_0("Error while deserializing name layout template.");
+        }
+
+        List<NameTemplate2_0> list = new ArrayList<>();
+        list.add(nameTemplate);
+        this.setNameTemplate(list);
+    }
+
+    @Override
+    public void globalPropertyDeleted(String s) {
+        if (getNameFormat().equals(s))
+            this.setNameTemplate((List)(new Vector()));
+    }
+
+    public String getNameFormat() {
+
+        return "<org.openmrs.module.webservices.rest.web.v1_0.template.openmrs2_0.NameTemplate2_0>" +
```

Comment:
Where did you get this from?

Reply:
That is the String for the "layout.name.format" value which didn't exist